compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -17,9 +17,6 @@ use std::collections::VecDeque;
 use std::iter;
 use std::ops::RangeFrom;
 
-const DEFAULT_THRESHOLD: usize = 50;
```

Comment:
Since there are lots more "constants" (and we may want to change them in the future), maybe we could create a more general solution: A way to pass a key/value pair list to specific optimizations. This way we don't have to (re)create a command line flag for every tiny thing, but can just parse arbitrary things out of the key/value pair list.

Reply:
What do you think about something like the above? Or do you have other ideas for generalizing configurable mir optimizations?

Reply:
We could also consider add new category of flags dedicated for MIR optimizations. I don't know if we need a completely different mechanism, especially with those flags being the first of a kind.

As far as enabling / disabling specific pass this might be better handled in context of #77665, so I could leave this out.

Reply:
> I don't know if we need a completely different mechanism, especially with those flags being the first of a kind.

That's fair. So we'll revisit once we get more than like 5 flags in total or sth.

> As far as enabling / disabling specific pass this might be better handled in context of #77665, so I could leave this out.

If your work is blocked on this, we can merge it now and remove it again later, but otherwise leaving it out for #77665 would be best. It may just take a bit for that to happen.

Reply:
Modified the PR to add threshold flag only for now.